pkg/operator/status.go

```diff
@@ -1,61 +1,49 @@
 package operator
 
 import (
-	"fmt"
-
-	"github.com/openshift/cluster-version-operator/lib/resourceapply"
-	"github.com/openshift/cluster-version-operator/pkg/apis/operatorstatus.openshift.io/v1"
+	osconfigv1 "github.com/openshift/api/config/v1"
+	osclientsetv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	cvoresourcemerge "github.com/openshift/cluster-version-operator/lib/resourcemerge"
 	"github.com/openshift/machine-api-operator/pkg/version"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
 )
 
-// syncStatus applies the new condition to the mco's OperatorStatus object.
-func (optr *Operator) syncStatus(cond v1.OperatorStatusCondition) error {
-	if cond.Type == v1.OperatorStatusConditionTypeDegraded {
-		return fmt.Errorf("invalid cond %s", cond.Type)
-	}
-
-	// TODO(yifan): Fill in the Extention field for the status
+//syncStatus applies the new condition to the mao ClusterOperator object.
+func (optr *Operator) syncStatus(cond osconfigv1.ClusterOperatorStatusCondition) error {
 	// to report the status of all the managed components.
-	status := &v1.OperatorStatus{
+	clusterOperator := &osconfigv1.ClusterOperator{
```

Comment:
What's the relation between mao operator status and cluster operator status? Is cluster operator an abstract way how to update status? Or WIP at the moment? If anything, I would expect `osconfigv1.Operator` struct instead for generic status updates.

Hmm, mao cluster operator. What do I need to read to understand why is the `cluster` part of mao operator name? Assuming it means a cluster operator is a name for a class of operators, right? On the cluster level?